wagtail/search/backends/database/mysql/mysql.py

```diff
@@ -344,9 +345,12 @@ def get_search_field(self, field_lookup, fields=None):
 
     def build_search_query_content(self, query, invert=False):
         if isinstance(query, PlainText):
-            terms = query.query_string.split()
+            # For Boolean full text search queries in MySQL,
+            # non-alphanumeric characters act as separators
+            terms = [term for term in re.split("\W+", query.query_string) if term]
```

Comment:
I get the following warning when I ran the bakerydemo setup with MySQL:

```
/code/wagtail/wagtail/search/backends/database/mysql/mysql.py:350: SyntaxWarning: invalid escape sequence '\W'
  terms = [term for term in re.split("\W+", query.query_string) if term]
```

This should be

```suggestion
            terms = [term for term in re.split(r"\W+", query.query_string) if term]
```

as the `\` need to be escaped, so either `r"\W+"` or `"\\W+"`

Reply:
Oops - good catch!